components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/util/ResponseTypeHandlerUtil.java

```diff
@@ -716,10 +715,20 @@ private static long getConfiguredAccessTokenValidityPeriodInMillis(OAuthAuthzReq
         return validityPeriodInMillis;
     }
 
-    private static long getConfiguredRefreshTokenValidityPeriodInMillis(OAuthAppDO oAuthAppBean) {
+    private static long getConfiguredRefreshTokenValidityPeriodInMillis(OAuthAppDO oAuthAppBean,
+                                                                        OAuthAuthzReqMessageContext oauthAuthzMsgCtx) {
 
         long refreshTokenValidityPeriodInMillis;
-        if (oAuthAppBean.getRefreshTokenExpiryTime() != 0) {
+        long callbackValidityPeriod = oauthAuthzMsgCtx.getRefreshTokenvalidityPeriod();
```

Comment:
Can we use a better variable name?

callbackValidityPeriod does not give the intended meaning.